nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java

```diff
@@ -87,12 +87,15 @@ public boolean onOverrideMayBeNullExpr(
       @Nullable Symbol exprSymbol,
       VisitorState state,
       boolean exprMayBeNull) {
-    if (expr.getKind().equals(Tree.Kind.METHOD_INVOCATION)
+    if (exprMayBeNull) {
+      return true;
+    }
+    if (expr.getKind() == Tree.Kind.METHOD_INVOCATION
         && exprSymbol instanceof Symbol.MethodSymbol
         && optionalIsGetCall((Symbol.MethodSymbol) exprSymbol, state.getTypes())) {
       return true;
```

Comment:
in the old code, this line returning true (and thus the whole if condition) only matters, if we wouldnt eventually be returning true with `return exprMayBeNull` anyway.

this again is assuming that the checks have no side-effects (if they do tests around it are missing)